core/admin/EE_Admin_Page_Loader.core.php

```diff
@@ -335,21 +342,23 @@ private function _get_installed_pages()
                     $caf_admin_page = $this->_caffeinated_extends[ $this->_current_caf_extend_slug ]['admin_page'];
                     add_filter(
                         "FHEE__EE_Admin_Page_Init___initialize_admin_page__path_to_file__{$menu_slug}_{$admin_page_name}",
-                        function ($path_to_file) use ($caf_path) {
+                        static function ($path_to_file) use ($caf_path) {
                             return $caf_path;
                         }
                     );
                     add_filter(
                         "FHEE__EE_Admin_Page_Init___initialize_admin_page__admin_page__{$menu_slug}_{$admin_page_name}",
-                        function ($admin_page) use ($caf_admin_page) {
+                        static function ($admin_page) use ($caf_admin_page) {
                             return $caf_admin_page;
                         }
                     );
                     $extend = true;
                 }
                 // let's do the registered hooks
                 $extended_hooks = $this->_installed_pages[ $page ]->register_hooks($extend);
-                $hooks_ref = array_merge($hooks_ref, $extended_hooks);
+                foreach ($extended_hooks as $key => $value) {
+                    $hooks_ref[] = $value;
```

Comment:
should not run `array_merge()` inside loops

Reply:
it's not about references it's because `array_merge()` makes copies of all of the arrays which can just eat up system resources when you are doing so inside a loop, so it is not recommended. Outside of a loop it's ok but it does take significantly longer than just adding elements via a loop like I have done. How you do that depends on the the data and what keys there are. These arrays were numerically indexed so what I did is fine. Basically, any time i see simple optimizations like this I go ahead and do them because every little bit adds up over time when we are running a platform like ES.